src/instruction.cpp

```diff
@@ -59,14 +59,27 @@ instruction::instruction(literal l)
 
 void instruction::replace(const shape& r)
 {
-    if(r != result)
-    {
+    if (r != result){
         result = r;
-        for(auto&& ins : output)
+        std::deque<instruction_ref> q;
+        for (instruction_ref ins: output){
+            q.push_back(ins);
+        }
```

Comment:
Initialize the `deque` with the `output`:  `std::deque<instruction_ref> q(output.begin(), output.end())`.

Reply:
Fixed